gui/wxpython/startup/locdownload.py

```diff
@@ -505,11 +604,14 @@ def OnClose(self, event):
             ret = dlg.ShowModal()
             dlg.Destroy()
 
-            # TODO: terminate download process on wx.ID_YES
             if ret == wx.ID_NO:
                 return
+            else:
+                self.panel.thread.Terminate()
+                self.panel._change_download_btn_label()
 
-        self.Close()
+        if event:
+            self.Close()
```

Comment:
This works for Mac: `self.EndModal(wx.ID_CANCEL)` instead of `self.Close()`

Reply:
What about this? This addresses the more serious issue on mac.

Reply:
Yes this is working properly. Thanks, I've already fixed it.